README.md

```diff
@@ -1,42 +1,144 @@
+| `postfix_saslauthd_options`           |     See [#saslauthd]      | saslauthd configuration options, see [#saslauthd]                                 |
+
+## Postfix Configuration
+
+The `postfix_configs` variable is used to configure Postfix configuration files under
+`/etc/postfix/`.
+The two most important configuration files are `main.cf` and `master.cf` (see Postfix
+documentation), but `postfix_configs` can specify arbitrary additional config files.
+Example:
+
+```yml
+postfix_configs:
+  main: |
+    # Content of /etc/postfix/main.cf goes here
+  master: |
+    # Content of /etc/postfix/main.cf goes here
+  arbitrary-additional-config: |
+    # Content of /etc/postfix/arbitrary-additional-config.cf goes here
+```
 
-Don't forget to indent the markdown table so it is readable even if not rendered.
+As displayed in the example, `postfix_configs` is a dict that maps file names to file contents.
+Those files are placed in `/etc/postfix/` and file names are extended by `.cf`.
 
-| Name       | Required/Default         | Description                                                                                        |
-|------------|:------------------------:|----------------------------------------------------------------------------------------------------|
-| `example1` | :heavy_check_mark:       | Lorem ipsum dolor sit amet, consetetur sadipscing elitr,                                           |
-| `example2` | :heavy_multiplication_x: | Sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. |
-| `example3` | `True`                   | Stet clita kasd gubergren                                                                          |
-| `example4` | `5`                      | No sea takimata sanctus est Lorem ipsum dolor sit amet.                                            |
+Any surplus config files `/etc/postfix/*.cf` are removed by this role, except for the following
+files:
 
+- `main.cf`
+- `master.cf`
+- `dynamicmaps.cf`
 
-## Example
+These are left untouched in case they are missing from `postfix_configs`, in order to allow using
+distribution-supplied defaults for these files.
 
-Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+## saslauthd
+
+If you need to authenticate users against an LDAP directory, this roles can install and enable
+saslauthd.
+This can be enabled by setting `postfix_sasl_enabled: True`.
+
+If you need bind credentials or other additional configuration options, you need to set
+`postfix_saslauthd_options`.
+This dict will be parsed and written to the saslauthd config file.
+The default is:
 
 ```yml
+postfix_saslauthd_options:
+  ldap_servers: "{{ postfix_ldap_servers }}"
+  ldap_search_base: "{{ postfix_ldap_search_base }}"
+  ldap_auth_method: bind
 ```
```

Comment:
I removed the default and phrased it as an example instead. There was no reason to have these `postfix_ldap_*` variables, because they were only used in this default. Instead one can directly configure `postfix_saslauthd_options`.